src/ui/common/src/components/integrations/dialogs/gcsDialog.tsx

```diff
@@ -0,0 +1,67 @@
+import { Checkbox, FormControlLabel } from '@mui/material';
+import Box from '@mui/material/Box';
+import React, { useEffect, useState } from 'react';
+
+import { GCSConfig, IntegrationConfig } from '../../../utils/integrations';
+import { IntegrationTextInputField } from './IntegrationTextInputField';
+
+const Placeholders: GCSConfig = {
+  bucket: 'aqueduct',
+  credentials_path: '',
+  use_as_storage: '',
+};
+
+type Props = {
+  setDialogConfig: (config: IntegrationConfig) => void;
+};
+
+export const GCSDialog: React.FC<Props> = ({ setDialogConfig }) => {
+  const [bucket, setBucket] = useState<string>(null);
+  const [credentialsPath, setCredentialsPath] = useState<string>(null);
+  const [useAsStorage, setUseAsStorage] = useState<string>('false');
+
+  useEffect(() => {
+    const config: GCSConfig = {
+      bucket: bucket,
+      credentials_path: credentialsPath,
+      use_as_storage: useAsStorage,
+    };
+    setDialogConfig(config);
+  }, [bucket, credentialsPath, useAsStorage]);
+
+  return (
+    <Box sx={{ mt: 2 }}>
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={true}
+        label="Bucket*"
+        description="The name of the GCS bucket."
+        placeholder={Placeholders.bucket}
+        onChange={(event) => setBucket(event.target.value)}
+        value={bucket}
+      />
+
+      <IntegrationTextInputField
```

Comment:
@vsreekanti Currently the only way to provide credentials is via a filepath. Do you think we should prioritize allowing users to upload a file as well (the same we do for connecting to s3)?

Reply:
Added the file upload method as we discussed